### PR TITLE
Support deploying nova via the top level Nova CRD

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

**Comment by gibizer** [comment]
This now works with the following settings
export NOVA_REPO=https://github.com/gibizer/nova-operator.git
export NOVA_BRANCH=nova-top-level-crd
export NOVA_IMG=quay.io/gibi/nova-operator-index:v0.0.2

File: Makefile [hunk 1]
```diff
@@ -499,3 +505,46 @@ ironic_deploy_cleanup: ## cleans up the service instance, Does not affect the op
...
+	bash scripts/operator-cleanup.sh
+	rm -Rf ${OPERATOR_DIR}
+
+.PHONY: nova_deploy_prep
+nova_deploy_prep: export KIND=Nova
+# TOOD(gibi): the tooling expect a containerImage at the top level
+# but for projects like Cinder and Nova where there are multiple services with
+# different images this customization does not make sense. Make this
+# customization optional in the tooling.
+nova_deploy_prep: export IMAGE="unused"
```
Comment (SeanMooney) [inline Makefile:533]: this is the image of the operator
there will only be one operator image for nova so i think your got confused with the container images used by the nova binary contianers

Comment (SeanMooney) [reply Makefile:533]: oh never mind this is the deploy one ya hova will have several

**Comment by abays** [comment]
> This now works with the following settings export NOVA_REPO=https://github.com/gibizer/nova-operator.git export NOVA_BRANCH=nova-top-level-crd export NOVA_IMG=quay.io/gibi/nova-operator-index:v0.0.2

Without these overrides, this depends on https://github.com/openstack-k8s-operators/nova-operator/pull/61?

**Comment by gibizer** [comment]
> > This now works with the following settings export NOVA_REPO=https://github.com/gibizer/nova-operator.git export NOVA_BRANCH=nova-top-level-crd export NOVA_IMG=quay.io/gibi/nova-operator-index:v0.0.2
> 
> Without these overrides, this depends on [openstack-k8s-operators/nova-operator#61](https://github.com/openstack-k8s-operators/nova-operator/pull/61)?

Yes the it depends on the [openstack-k8s-operators/nova-operator#61](https://github.com/openstack-k8s-operators/nova-operator/pull/61) PR to have the CRDs defined.

**Review by stuggi — APPROVED** [review]
/lgtm

**Review by dprince — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -565,3 +572,48 @@ octavia_deploy_cleanup: ## cleans up the service instance, Does not affect the o
...
+	$(eval $(call vars,$@,nova))
+	mkdir -p ${OPERATOR_BASE_DIR} ${OPERATOR_DIR} ${DEPLOY_DIR}
+	pushd ${OPERATOR_BASE_DIR} && git clone -b ${NOVA_BRANCH} ${NOVA_REPO} && popd
+	cp ${NOVA_CR} ${DEPLOY_DIR}
+	bash scripts/gen-service-kustomize.sh
+
+.PHONY: nova_deploy
+nova_deploy: input nova_deploy_prep ## installs the service instance using kustomize. Runs prep step in advance. Set NOVA_REPO and NOVA_BRANCH to deploy from a custom repo.
+	$(eval $(call vars,$@,nova))
+	oc kustomize ${DEPLOY_DIR} | oc apply -f -
```
Comment (SeanMooney) [inline Makefile:611]: by the way i think 
oc kustomize ${DEPLOY_DIR} | oc apply -f -
could be 
oc apply -k ${DEPLOY_DIR} but i have not tried that.

kustomize is integrated into kubctl/oc in more recent version i belive so we likely can simplify this in the future if we want too.

**Pull request merged**